sidechain/src/tests.rs

```diff
@@ -26,259 +26,220 @@ fn get_signer(pubkey: &[u8; 32]) -> AccountId {
 }
 
 #[test]
-fn confirm_proposed_sidechain_block_works_for_correct_shard() {
+fn confirm_imported_sidechain_block_works_for_correct_shard() {
 	new_test_ext().execute_with(|| {
 		Timestamp::set_timestamp(TEST7_TIMESTAMP);
-		let block_hash = H256::default();
+		let hash = H256::default();
 		let signer7 = get_signer(TEST7_SIGNER_PUB);
 		let shard7 = H256::from_slice(&TEST7_MRENCLAVE);
 
-		register_enclave();
-
-		let header = SidechainHeader {
-			parent_hash: block_hash,
-			block_number: 1,
-			shard_id: shard7,
-			block_data_hash: block_hash,
-		};
+		register_enclave7();
 
-		assert_ok!(Sidechain::confirm_proposed_sidechain_block(
+		assert_ok!(Sidechain::confirm_imported_sidechain_block(
 			Origin::signed(signer7.clone()),
 			shard7.clone(),
-			header.clone(),
+			1,
+			hash
 		));
 
 		let expected_event =
-			Event::Sidechain(SidechainEvent::ProposedSidechainBlock(signer7, block_hash));
+			Event::Sidechain(SidechainEvent::ImportedSidechainBlock(signer7, hash));
```

Comment:
Hmm, I think I would pick another name for the event, as we don't import any blocks here.

What would you think about:
* SidechainBlockFinalized/FinalizedSidechainBlock

Sounds a bit more epic and it is actually what is happening.

Reply:
Thanks for the input, I also adjusted one of the function names accordingly. Fits much better now :-)